lib/Graph/Nodes.cpp

```diff
@@ -200,7 +200,11 @@ static bool verifyPool(NodeValue src, NodeValue dest,
   ShapeNHWC exp(idim.n, outSz.first, outSz.second, idim.c);
   isValid &=
       expectCompareTrue("Unexpected output dimensions", exp, odim, parent);
-  isValid &= checkTypeIgnoreShape(src, dest, dest.getNode());
+  if (src.getType()->isQuantizedType()) {
+    isValid &= checkSameIsQuantized(src.getType(), dest.getType(), parent);
```

Comment:
this check is a bit relaxed. For maxpool we'd like to check that type with ignored shape match as well (not only that the input/output is quantized).

Reply:
slightly simplified comparison.
both needs to be either quantized or not quantized and in case of max pool we'd need to check matching type except shape.

``` cpp
isValid &= checkSameIsQuantized(src.getType(), dest.getType(), parent);

if (!isAvgPool) {
  isValid &= checkTypeIgnoreShape(src, dest, parent);
}
```